lib/rules/require-await.js

```diff
@@ -42,14 +42,41 @@ module.exports = {
         schema: [],
 
         messages: {
-            missingAwait: "{{name}} has no 'await' expression."
-        }
+            missingAwait: "{{name}} has no 'await' expression.",
+            removeAsync: "Remove 'async'."
+        },
+
+        hasSuggestions: true
     },
 
     create(context) {
         const sourceCode = context.sourceCode;
         let scopeInfo = null;
 
+        /**
+         * Finds the range that covers the `async` keyword.
+         * @param {FunctionNode} node The function node to find the `async` keyword in.
+         * @returns {[number, number]} The range of the `async` keyword.
+         */
+        function findAsyncKeywordRange(node) {
+
+            /*
+             * If the function belongs to a method definition or
+             * property, then the function's range may not include the
+             * `async` keyword and we should look at the parent instead.
+             */
+            const nodeWithAsyncKeyword =
+                (node.parent.type === "MethodDefinition" && node.parent.value === node) ||
+                (node.parent.type === "Property" && node.parent.method && node.parent.value === node)
+                    ? node.parent
+                    : node;
+
+            const asyncToken = sourceCode.getFirstToken(nodeWithAsyncKeyword, token => token.value === "async");
+
+
+            return [asyncToken.range[0], sourceCode.getTokenAfter(asyncToken).range[0]];
```

Comment:
Shouldn't we only remove `async` and leave space alone? Currently we will remove comments

Reply:
We could use just the range of the `async` keyword and extend it by one character if the next character is a space. 

I'll add a test with a comment after the `async` keyword to verify that it won't end up being removed (it currently is removed, as you point out).